### PR TITLE
MEN-3978: Rename environment tools to grub-mender-grubenv-*.

### PR DESCRIPTION
**Pull request by kacf**
This renames the tools, adjusts the install scripts, and adds a test
for the Makefile installation. There is also a new target,
"install-legacy-tools", which installs the tools using the old names.

Changelog: User space tools, "fw_printenv" and "fw_setenv" have been
renamed to "grub-mender-grubenv-print" and "grub-mender-grubenv-set",
respectively. This has been done in order to avoid conflict with
U-Boot's user space tools, which have the same names.

Signed-off-by: Kristian Amlie <kristian.amlie@northern.tech>

### CODE REVIEW

**Comment by kacf** [comment]
This requires followups in the client and in meta-mender, but it's a start.

**Review by drewmoseley — APPROVED** [review]

File: tests/test_makefile.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+find "$TMPDIR/install" -type f -o -type l | sed -e "s,^$TMPDIR/install/,," | sort > "$TMPDIR/actual.log"
+cat > "$TMPDIR/expected.log" <<EOF
+boot/efi/EFI/BOOT/grub.cfg
+boot/efi/EFI/BOOT/mender_grubenv1/env
+boot/efi/EFI/BOOT/mender_grubenv1/lock
+boot/efi/EFI/BOOT/mender_grubenv1/lock.sha256sum
+boot/efi/EFI/BOOT/mender_grubenv2/env
+boot/efi/EFI/BOOT/mender_grubenv2/lock
+boot/efi/EFI/BOOT/mender_grubenv2/lock.sha256sum
```
Comment (lluiscampos) [inline tests/test_makefile.sh:29]: By inspecting the Makefile, I don't understand why we don't expect these in the `install-legacy-tools` test case.

**Review by lluiscampos — CHANGES_REQUESTED** [review]
The test is not running, it silently fails in the `make` command:
```
++ sed -ne '/^mender_rootfsa_part=/ {s/^mender_rootfsa_part=//; p}' mender_grubenv_defines
sed: can't read mender_grubenv_defines: No such file or directory
```

I also run into the same error locally when trying to understand better the test.

**Comment by lluiscampos** [comment]
> The test is not running (...)

By the way, nice idea adding this test! :guitar:

File: tests/test_makefile.sh (hunk 1 above)
Comment (kacf) [reply tests/test_makefile.sh:29]: The first test is installing everything, whereas the second is only installing the tools. I suppose I could make an additional case which only installs the regular tools, to make this a bit clearer.

**Comment by kacf** [comment]
> The test is not running, it silently fails in the `make` command:

Well spotted! I think this is not important to the test, but it should be fixed anyway. I'll definitely take a look at it.

File: tests/test_makefile.sh (hunk 1 above)
Comment (lluiscampos) [reply tests/test_makefile.sh:29]: Ah, of course. Got it.

I don't think we need to add an extra test for that.

**Review by mirzak — APPROVED** [review]

**Comment by kacf** [comment]
> > The test is not running, it silently fails in the `make` command:
> 
> Well spotted! I think this is not important to the test, but it should be fixed anyway. I'll definitely take a look at it.

It only took seven months, but at last I have fixed this! :-)

**Review by lluiscampos — APPROVED** [review]

**Pull request merged**